modules/asm/main.tf

```diff
@@ -14,95 +14,57 @@
  * limitations under the License.
  */
 
-data "google_project" "asm_project" {
-  project_id = var.project_id
+locals {
+  // GKE release channel is a list with max length 1 https://github.com/hashicorp/terraform-provider-google/blob/9d5f69f9f0f74f1a8245f1a52dd6cffb572bbce4/google/resource_container_cluster.go#L954
+  gke_release_channel = data.google_container_cluster.asm.release_channel != null ? data.google_container_cluster.asm.release_channel[0].channel : ""
```

Comment:
This does not handle the case where the GKE module returns "unspecified" as the release channel. It results in the same string being passed to the cpr module.

Reply:
Fixed, thanks.